### PR TITLE
Pass sparsity argument directly to `extract_waveforms`

### PR DESCRIPTION
**Pull request by alejoe91**

### CODE REVIEW

File: spikeinterface/sortingcomponents/peak_pipeline.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ class WaveformExtractorNode(PipelineNode):
     """Base class for waveform extractor"""
 
     def __init__(self, recording: BaseRecording, 
-                 ms_before: float, ms_after: float, parents: Optional[list[PipelineNode]]=None, return_ouput: bool=True):
+                 ms_before: float, ms_after: float, parents: Optional[List[PipelineNode]]=None, return_ouput: bool=True):
```
Comment (alejoe91) [inline spikeinterface/sortingcomponents/peak_pipeline.py:59]: This is needed for Python3.8 (and also work for > versions). Using `list[]` directly is not supported in 3.8

Comment (h-mayorquin) [reply spikeinterface/sortingcomponents/peak_pipeline.py:59]: So they only stared making this available after python 3.9? Good to know.

File: spikeinterface/core/waveform_extractor.py [hunk 2]
```diff
@@ -1424,11 +1427,18 @@ def extract_waveforms(recording, sorting, folder=None,
...
+                                           num_spikes_for_sparsity=num_spikes_for_sparsity,
+                                           **estimate_kwargs, **job_kwargs)
+        else:
+            assert isinstance(sparsity, ChannelSparsity), "'sparsity' must be a ChannelSparsity object"
+            unit_id_to_channel_ids = sparsity.unit_id_to_channel_ids
+            assert all(u in sorting.unit_ids for u in unit_id_to_channel_ids), \
+                    "Invalid unit ids in sparsity"
+            for channels in unit_id_to_channel_ids.values():
+                assert all(ch in recording.channel_ids for ch in channels), \
+                    "Invalid channel ids in sparsity"
```
Comment (samuelgarcia) [inline spikeinterface/core/waveform_extractor.py:1441]: I think we should have something more easy.
In short `extract_waveforms(recording, sorting, sparsity=something)` should be possible.
Here we force `extract_waveforms(recording, sorting, sparsity=something, sprase=True)`

In my mind `sprse=True` is "try to make make it sparse with estimation"

Comment (alejoe91) [reply spikeinterface/core/waveform_extractor.py:1441]: Ok, so if you pass sparsity `sparse` is ignored ;) I agree

File: spikeinterface/sortingcomponents/peak_pipeline.py (hunk 1 above)
Comment (alejoe91) [reply spikeinterface/sortingcomponents/peak_pipeline.py:59]: yeah, it was failing on 3.8 envs. Using `List` seems to be compatible to both

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**